tests/ui/for_loop.rs

```diff
@@ -550,6 +550,15 @@ pub fn manual_copy(src: &[i32], dst: &mut [i32], dst2: &mut [i32]) {
     for i in 0..10 {
         dst_vec[i] = src[i];
     }
+
+    // Copying entire source slice - simplify suggestion (issue #3004)
+    let src = [0, 1, 2, 3, 4];
+    let mut dst = [0, 0, 0, 0, 0, 0];
+    let from = 1;
+
+    for i in from..from + src.len() {
```

Comment:
Please also add a test for values other than `src.len()`. I think your PR will also improve the suggestion in that case